jarviscli/plugins/teachme.py

```diff
@@ -0,0 +1,420 @@
+        for document in corpus_words:
+            ClearedText = []
+            for i in document:
+                if i.isalpha():
+                    if check_entity(i, sp2):
+                        lemmatized, pos = get_lemma(i, sp)
+                        if Forbidden_List.get(lemmatized, True) and not is_not_punctuation(i):
+                            ClearedText.append(lemmatized)
+                            words.append(lemmatized)
+                            if lemmatized not in corpus_words_dic:
+                                corpus_words_dic[lemmatized] = [pos]
+            if len(ClearedText) > 0:
+                clean_corpus_words.append(ClearedText)
+        number_docs = len(clean_corpus_words)
+        document_count = {}
+        idfs = []
+        for document in clean_corpus_words:
+            word_set = np.unique(document)
+
+            for word in word_set:
+                document_count[word] = document_count.get(word, 0) + 1
+
+        for word in document_count:
+            idf = np.log(number_docs / document_count[word])
+            corpus_words_dic[word].append(idf)
+            idfs.append(idf)
+        xarray = np.array(idfs)
+        mn = np.min(xarray)
+        mx = np.max(xarray)
+        rng = mx - mn
+        idfs = list((xarray - mn) / rng)
+        for word in corpus_words_dic:
+            corpus_words_dic[word][1] = ((corpus_words_dic[word][1] - mn) / rng)
+        del (clean_corpus_words)
+        dist = Counter(idfs)
+        dist = list(dist.items())
+        dist.sort(key=lambda x: x[0])
+        dist = np.array(dist)
+        norm = np.dot(dist.T[0], dist.T[1])
+        del (idfs)
+        return words, corpus_words_dic, dist, norm
+
+    def get_word_out(word, sentence, sp):
+        g = wt(sentence)
+        l = [get_lemma(i, sp)[0] for i in g]
+        inx = l.index(word)
+        OrgWord = g[inx]
+        g[inx] = '_' * len(OrgWord)
+        question = " ".join([j for j in g])
+        question = question + '.'
+        return question, OrgWord
+
+    def syn_ant(word):
+        from nltk.corpus import wordnet
+        synonyms = []
+        antonyms = []
+        for syn in wordnet.synsets(word):
+            for l in syn.lemmas():
+                synonyms.append(l.name())
+                if l.antonyms():
+                    antonyms.append(l.antonyms()[0].name())
+        synon = ''
+        ant = ''
+        for i in set(synonyms):
+            synon = synon + i.capitalize() + ', '
+        for i in set(antonyms):
+            ant = ant + i.capitalize() + ', '
+        synon = synon + ' '
+        synon = synon.replace(',  ', '.')
+        ant = ant + ' '
+        ant = ant.replace(',  ', '.')
+        if synon == '  ':
+            synon = 'No synonym is found for "' + word.capitalize() + '".'
+        if ant == ' ':
+            ant = 'No antononym is found for "' + word.capitalize() + '".'
+        return synon, ant, synonyms
+
+    def get_def_oxford(word, app_id, app_key):

```

Comment:
Question: Would it be possible to (optionally) skip definition or use another (maybe less good, but free) source?

Reply:
would words API https://www.wordsapi.com/ be an appropriate source @AurelienBuchet

Reply:
I'll have a look at it with the other, it should be possible to have to choice between several options. By the way it is possible to register for an Oxford API key with a prototype account for free.

Reply:
Hi all, 
Thank you for the suggestion to add other APIs. As you know, each APIs has a specific structure. "Teach me" currently supports only Oxford API that offers a free account. It is worth mentioning that, Oxford API possesses a lot of information regarding the word such as etymology and examples in different contexts. However, we will consider other powerful APIs in future commits. It is easy as a piece of a cake, but takes time, 

Best,

Nima